core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/SubmissionApiMetadataParamsSpec.scala

```diff
@@ -0,0 +1,248 @@
+package org.broadinstitute.dsde.rawls.webservice
+
+import akka.http.scaladsl.model.Uri
+import akka.http.scaladsl.server.Route.{seal => sealRoute}
+import org.broadinstitute.dsde.rawls.dataaccess._
+import org.broadinstitute.dsde.rawls.google.MockGooglePubSubDAO
+import org.broadinstitute.dsde.rawls.model.ExecutionJsonSupport.MetadataParamsFormat
+import org.broadinstitute.dsde.rawls.model._
+import org.broadinstitute.dsde.rawls.openam.MockUserInfoDirectives
+import spray.json._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+ * Created by davidan on 11/5/18.
+ */
+class SubmissionApiMetadataParamsSpec extends ApiServiceSpec {
+
+  class ParamValidatingExecutionServiceDAO(executionServiceURL: String, override val workbenchMetricBaseName: String) extends HttpExecutionServiceDAO(executionServiceURL: String,  workbenchMetricBaseName: String) {
+    override def callLevelMetadata(id: String, metadataParams: MetadataParams, userInfo: UserInfo): Future[JsObject] = {
+      // returns the MetadataParams from argument list, so tests can validate them
+      Future.successful(metadataParams.toJson.asJsObject)
+    }
+  }
+
+  case class TestApiService(dataSource: SlickDataSource, gcsDAO: MockGoogleServicesDAO, gpsDAO: MockGooglePubSubDAO, override val executionServiceCluster: MockShardedExecutionServiceCluster)
+                           (implicit override val executionContext: ExecutionContext) extends ApiServices with MockUserInfoDirectives
+
+
+  def withApiServices[T](dataSource: SlickDataSource, gcsDAO: MockGoogleServicesDAO = new MockGoogleServicesDAO("test"))(testCode: TestApiService =>  T): T = {
+    val apiService = new TestApiService(dataSource, gcsDAO, new MockGooglePubSubDAO, MockShardedExecutionServiceCluster.fromDAO(new ParamValidatingExecutionServiceDAO("unused", "unused"), dataSource))
+
+    try {
+      testCode(apiService)
+    } finally {
+      apiService.cleanupSupervisor
+    }
+  }
+
+  def withTestDataApiServices[T](testCode: TestApiService =>  T): T = {
+    withDefaultTestDatabase { dataSource: SlickDataSource =>
+      withApiServices(dataSource)(testCode)
+    }
+  }
+
+  val submissionId = testData.submission1.submissionId
+  val workflowId = testData.submission1.workflows.head.workflowId.get // assume it exists; if it doesn't, will throw exception so the test fails
+
+  val basePath = s"${testData.workspace.path}/submissions/$submissionId/workflows/$workflowId"
+
+  "SubmissionApi" should "use defaults for metadata params when no querystring" in withTestDataApiServices { services =>
+    Get(s"$basePath") ~>
+      sealRoute(services.submissionRoutes) ~>
+      check {
+        val actualParams = responseAs[MetadataParams]
+        // get the defaults
+        val expectedParams = new MetadataParams()
```

Comment:
You don't need to `new` for case class

Reply:
thanks for the catch - I dunno why I used that here!